### PR TITLE
[CLEANUP] avoid private API for computed properties in tests

### PR DESCRIPTION
**Pull request by pangratz**
This addresses #3936

cc @krisselden

### CODE REVIEW

File: addon/-private/system/relationships/ext.js [hunk 1]
```diff
@@ -12,8 +12,8 @@ var Map = Ember.Map;
 var MapWithDefault = Ember.MapWithDefault;
 
 var relationshipsDescriptor = Ember.computed(function() {
-  if (Ember.testing === true && relationshipsDescriptor._cacheable === true) {
-    relationshipsDescriptor._cacheable = false;
+  if (Ember.testing === true && relationshipsDescriptor._volatile === true) {
```
Comment (sly7-7) [inline addon/-private/system/relationships/ext.js]: should'nt the condition be inversed ? Per @krisselden comment: "Should not use relatedTypesDescriptor._cacheable = false; in testing as this is now _volatile = true"

Comment (pangratz) [inline addon/-private/system/relationships/ext.js]: Ouch, wonderful catch! Thanks @sly7-7!!

Comment (sly7-7) [inline addon/-private/system/relationships/ext.js]: Hum... seems not to be so wonderfull, the tests are now failing... I'm not sure I'm understand, or something else should be changed too ?

Comment (pangratz) [inline addon/-private/system/relationships/ext.js]: It looks like updating the tests as well helps with a happy Travis :wink: Thanks again for your :eyes:!

File: addon/-private/system/relationships/ext.js [hunk 2]
```diff
@@ -39,8 +39,8 @@ var relationshipsDescriptor = Ember.computed(function() {
 }).readOnly();
 
 var relatedTypesDescriptor = Ember.computed(function() {
-  if (Ember.testing === true && relatedTypesDescriptor._cacheable === true) {
-    relatedTypesDescriptor._cacheable = false;
+  if (Ember.testing === true && relatedTypesDescriptor._volatile === true) {
```
Comment (sly7-7) [inline addon/-private/system/relationships/ext.js]: so I think this one has been missed :eyeglasses: ?

Comment (pangratz) [inline addon/-private/system/relationships/ext.js]: :weary:, next :beer: is on me!

File: tests/integration/relationships/belongs-to-test.js [hunk 3]
```diff
@@ -563,15 +563,15 @@ test("relationshipsByName is cached in production", function(assert) {
   //We set the cacheable to true because that is the default state for any CP and then assert that it
```
Comment (sly7-7) [inline tests/integration/relationships/belongs-to-test.js]: maybe the comment could be switched too ?

File: tests/integration/relationships/belongs-to-test.js [hunk 4]
```diff
@@ -581,15 +581,15 @@ test("relatedTypes is cached in production", function(assert) {
...
```
Comment (sly7-7) [inline tests/integration/relationships/belongs-to-test.js]: ^^

File: tests/integration/relationships/belongs-to-test.js [hunk 5]
```diff
@@ -599,15 +599,15 @@ test("relationships is cached in production", function(assert) {
...
```
Comment (sly7-7) [inline tests/integration/relationships/belongs-to-test.js]: ^^

File: tests/integration/relationships/belongs-to-test.js (hunk 3 above)
Comment (pangratz) [inline tests/integration/relationships/belongs-to-test.js]: :weary:, next :100: :beers: are on me!

Comment (sly7-7) [inline tests/integration/relationships/belongs-to-test.js]: Now I'm drunk :astonished:

**Comment by fivetanley** [comment]
I'm not sure what the original code is trying to accomplish. While this seems fine from a cleanup perspective, I wonder if we could remove it entirely. It seems to be only set in testing which could possibly lead to confusing bugs that don't show up in our test suite but might show up in an app's test suite.

**Comment by fivetanley** [comment]
@igorT do you know what's going on in the original code?

**Comment by krisselden** [comment]
@fivetanley I agree with you on this, I have no idea what this is doing, or why it would be passing now.   

Also cleanup implies you are refactoring the need for private API, here you are just renaming the field.

Please remove this code instead as @fivetanley suggests, then it actually will be cleanup.

**Comment by stefanpenner** [comment]
> I'm not sure what the original code is trying to accomplish.

Might be worth investigating this before prescribing solutions.

If I recall, this was a cross ember version related performance fix. And the above method at the time was the appropriate solution (while still being backwards compat and all that).

I am unsure if this is still needed, maybe @igorT knows.

**Comment by fivetanley** [comment]
> Might be worth investigating this before prescribing solutions.

"I wonder" is less of a prescription and more of a suggestion for investigation. :stuck_out_tongue:

**Comment by igorT** [comment]
There are two parts to this:
1. In ember data tests, we mess with volatile to assert that we do cache it in production.
2. We cache relationship links in production code, as model relationship options are assumed immutable once accessed, however this may not be the case when running your tests, as you could be reopening your classes very often. 

@stefanpenner I don't quite recall this being a backwards compat only thing?

**Comment by krisselden** [comment]
I'm still not clear, can you not just use `Ember.cacheFor(obj, key)` public API instead to assert it has been cached?

The descriptor property here was changed by me because I thought it was safe to change being private and I wanted it to align with the setter which is `desc.volatile()`.  This change also did not seem to break any tests, so I'm very unclear what the PR is about but this should not be touching a private field.

**Comment by stefanpenner** [comment]
just checked, the code i was thinking of no longer exists. So my above comment is likely just a distraction.

**Comment by bmac** [comment]
It seems like if the change to `_cacheable`/`_volatile` isn't causing our tests to fail then we should be able to remove the test code that modifies that property since it doesn't appear to be adding much value.

**Comment by bmac** [comment]
@pangratz can this pr be closed?

**Comment by pangratz** [comment]
@bmac yeah, let's close and investigate if the tests can be removed as you suggested here in your comment above.

**Pull request closed without merging**